File: data/multilang_comparison/program_c/src/func26.c
#include "program_c.h" 
array_t* func26(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t* array78;
      if (pCounter > 0) {
         array78 = vars->data[--pCounter];
         array78->refC++;
         DEBUG_COPY(array78->id);
      } else {
         array78 = (array_t*)malloc(sizeof(array_t));
         array78->size = 924;
         array78->refC = 1;
         array78->id = 78;
         array78->data = (unsigned int*)malloc(array78->size*sizeof(unsigned int));
         memset(array78->data, 0, array78->size*sizeof(unsigned int));
         DEBUG_NEW(array78->id);
      }
      unsigned int loop25 = 0;
      unsigned int loopLimit25 = (50)/3 + 1;
      for(; loop25 < loopLimit25; loop25++) {
         for (int i = 0; i < array78->size; i++) {
            array78->data[i]--;
         }
         for (int i = 0; i < array78->size; i++) {
            if (array78->data[i] == 24) { 
               return array78;
            }
         }
      }
      array_t_param params0;
      params0.size = 1;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array78;
      array_t* array79 = func36(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array79->id);
      free(params0.data);
      array79->refC--;
      if(array79->refC == 0) {
         free(array79->data);
         free(array79);
         DEBUG_FREE(array79->id);
      }
      array78->refC--;
      if(array78->refC == 0) {
         free(array78->data);
         free(array78);
         DEBUG_FREE(array78->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array80 = func32(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array80->id);
      free(params0.data);
      array80->refC--;
      if(array80->refC == 0) {
         free(array80->data);
         free(array80);
         DEBUG_FREE(array80->id);
      }
   }
   array_t* array81;
   if (pCounter > 0) {
      array81 = vars->data[--pCounter];
      array81->refC++;
      DEBUG_COPY(array81->id);
   } else {
      array81 = (array_t*)malloc(sizeof(array_t));
      array81->size = 379;
      array81->refC = 1;
      array81->id = 81;
      array81->data = (unsigned int*)malloc(array81->size*sizeof(unsigned int));
      memset(array81->data, 0, array81->size*sizeof(unsigned int));
      DEBUG_NEW(array81->id);
   }
   return array81;
}

File: data/multilang_comparison/program_nim/src/func26.nim
proc func26*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var array174: ptr ArrayT
      if pCounter > 0:
         dec pCounter
         array174 = vars.data[pCounter]
         inc array174.refC
         debugCopy(array174.id)
      else:
         array174 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
         array174.size = 692
         array174.refC = 1
         array174.id = 174
         array174.data = cast[ptr UncheckedArray[uint32]](alloc(array174.size * sizeof(uint32)))
         zeroMem(array174.data, array174.size * sizeof(uint32))
         debugNew(array174.id)
      let loopLimit52 = (50) div (3) + 1
      for loop52 in 0 ..< loopLimit52:
         discard
         for i in 0 ..< array174.size:
            dec array174.data[i]
         for i in 0 ..< array174.size:
            if array174.data[i] == 4:
               return array174
      var params0: ArrayTParam
      params0.size = 1
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      params0.data[0] = array174
      var array175: ptr ArrayT = func36(addr params0, getPath(), loopsFactor)
      debugReturn(array175.id)
      dealloc(params0.data)
      dec array175.refC
      if array175.refC == 0:
         dealloc(array175.data)
         dealloc(array175)
         debugFree(array175.id)
      dec array174.refC
      if array174.refC == 0:
         dealloc(array174.data)
         dealloc(array174)
         debugFree(array174.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array176: ptr ArrayT = func32(addr params0, getPath(), loopsFactor)
      debugReturn(array176.id)
      dealloc(params0.data)
      dec array176.refC
      if array176.refC == 0:
         dealloc(array176.data)
         dealloc(array176)
         debugFree(array176.id)
   var array177: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array177 = vars.data[pCounter]
      inc array177.refC
      debugCopy(array177.id)
   else:
      array177 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array177.size = 429
      array177.refC = 1
      array177.id = 177
      array177.data = cast[ptr UncheckedArray[uint32]](alloc(array177.size * sizeof(uint32)))
      zeroMem(array177.data, array177.size * sizeof(uint32))
      debugNew(array177.id)
   return array177

